sdk/src/loom/Graphics/Shader.ls

```diff
@@ -0,0 +1,171 @@
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+===========================================================================
+*/
+
+package loom.graphics
+{
+    import loom2d.math.Matrix;
+    import loom2d.textures.Texture;
+
+    delegate ShaderBindDelegate();
+
+    /** Shader needs to be initialized after being created with 'load()' or 'loadFromAssets()'.
+      * When you are done using the shader, you should release the GPU resources by calling 'dispose()'.
+      *
+      * Every time the renderer binds the shader, 'onBind' delegte is invoked, where all needed uniforms
+      * should be set.
+      */
+
+    [Native(managed)]
+    public native class Shader
+    {
+        public native var MVP:Matrix;
+        public native var textureId:Number;
+
+        /**
+          * Creates a new Shader object. It needs to be initialized with
+          * load() or loadFromAssets().
+          */
+        public function Shader()
+        {
+        }
+
+        /**
+          * Releases GPU resources used by this shader.
+          */
+        public function dispose():void
+        {
+            deleteNative();
+        }
+
+        /**
+          * Loads the shader from strings. The parameters are plain source of shaders.
+          */
+        public native function load(vertexShader:String, fragmentShader:String):void;
+
+        /**
+          * Loads the shader from asset files.
+          */
+        public native function loadFromAssets(vertexShaderPath:String, fragmentShaderPath:String):void;
+
+        /**
+          * Returns the location of the named uniform. This value is then passed to 'setUniform' methods.
+          */
+        public native function getUniformLocation(name:String):Number;
+
+        /**
+          * Sets one 'float' uniform.
+          */
+        public native function setUniform1f(location:Number, v0:Number);
+
+        /**
+          * Sets an array of 'float' uniforms.
+          */
+        public native function setUniform1fv(location:Number, x1:Number, x2:Number, value:Vector.<Number>);
+
+        /**
+          * Sets one 'vec2' uniform
+          */
+        public native function setUniform2f(location:Number, v0:Number, v1:Number);
+
+        /**
+          * Sets an array of 'vec2' uniforms
+          */
+        public native function setUniform2fv(location:Number, value:Vector.<Number>);
+
+        /**
+          * Sets one 'vec3' uniform
+          */
+        public native function setUniform3f(location:Number, v0:Number, v1:Number, v2:Number);
+
+        /**
+          * Sets an array of 'vec3' uniforms
```

Comment:
Are there size requirements in the array? Are they enforced? (ie, %3 == 0?) Same goes for all of these that take an array.